cachi2/core/package_managers/pip.py

```diff
@@ -102,6 +110,47 @@ def fetch_pip_source(request: Request) -> RequestOutput:
         project_files=project_files,
     )
 
+# TODO _generate_purl func for multiple scenarios
+def _generate_purl(package: dict[str, Any]) -> str:
```

Comment:
For reference, in Cachito, we used to generate the purls for dependencies (not for the top-level package) like this https://github.com/containerbuildsystem/cachito/blob/6b01c763018f7569d2aeabbd982e2381297fc7cb/cachito/web/purl.py#L89

The difference in cachi2 is that we always want to use `pkg:pypi` (type=pypi) whereas Cachito used `pkg:github` or `pkg:generic` etc. for external dependencies

One particularly useful part might be the `checksum` qualifier for https dependencies. IMO we should include that in the purl until we add `hashes` to the SBOM